File: Reqnroll/Chapter14/14.4-TerminologyPromotionPattern/After/WIMP.Specs/Support/CustomParameterTypes.cs
using Reqnroll;

using WIMP.App.Models;

namespace WIMP.Specs.Support;

[Binding]
public class CustomParameterTypes
{
    [StepArgumentTransformation("(10\"|12\"|14\")", Name = "pizza-size")]
    public PizzaSize ConvertPizzaSize(string pizzaSizeString) =>
        pizzaSizeString switch
        {
            "10\"" => PizzaSize.Small,
            "12\"" => PizzaSize.Medium,
            "14\"" => PizzaSize.Large,
            _ => throw new FormatException($"Invalid size: {pizzaSizeString}")
        };
}

File: CucumberJVM/Chapter19/19.4-EmbeddedApplicationLogPattern/Before/src/test/java/com/wimp/app/specs/support/CustomParameterTypes.java
package com.wimp.app.specs.support;

import com.wimp.app.models.*;
import io.cucumber.java.ParameterType;

public class CustomParameterTypes {

    @ParameterType(value = "(10\"|12\"|14\")", name = "pizza-size")
    public PizzaSize convertPizzaSize(String pizzaSizeString) {
        return switch (pizzaSizeString) {
            case "10\"" -> PizzaSize.SMALL;
            case "12\"" -> PizzaSize.MEDIUM;
            case "14\"" -> PizzaSize.LARGE;
            default -> throw new IllegalArgumentException("Invalid size: " + pizzaSizeString);
        };
    }
}
